### ISSUE TITLE
Lab 04 Step 3

### ISSUE THREAD

**Issue by DrKBoucher** [Watts-College/paf-516-template#7]
@AntJam-Howell

I am having a problem when I get to the code chunk in step three.

URL <- "https://github.com/DS4PS/cpp-529-master/raw/master/data/ltdb_std_2010_sample.rds"
census.dat <- readRDS(gzcon(url( URL )))

msp <- merge( msp.pop1, census.dat, by.x="GEOID", by.y="tractid" )

From what I can see, census.dat does not have a GEOID column and msp.pop1 does not have tractid. Therefore, how can they merge? Am I just doing something wrong? 
Perhaps it might be in the previous code chunk where it says select( GEOID, estimate ). The only columns in msp.pop1 are GEOID, POP, and geometry. Should I be changing the variables within the parentheses?

**Comment by AntJam-Howell** [comment]
@DrKBoucher Did you check the data again?  Below I show the screenshot of what I see, showin the tractid column in census.dat. 

<img width="819" alt="Screenshot 2023-11-04 at 12 19 02 AM" src="https://github.com/Watts-College/paf-516-template/assets/51752063/3ec90545-9a57-4982-997b-75be8aaf9dab">

for San Diego MSA, the following code returns geoid. 

sd.pop <- get_acs( 
  geography = "tract", 
  variables = "B01003_001", 
  state = state.fips, 
  county = county.fips,
  geometry = TRUE ) %>% 
  select( GEOID, estimate ) %>%
  rename( POP=estimate )

**Comment by Jmurtos** [comment]
Hi, I'm having the same issue. 
Census.dat has tractid only while msp.pop1 has GEOID only. So after they merge, my new dataset has 0 rows. Not sure if I did something wrong.

**Comment by DrKBoucher** [comment]
> @DrKBoucher Did you check the data again? Below I show the screenshot of what I see, showin the tractid column in census.dat.
> 
> <img alt="Screenshot 2023-11-04 at 12 19 02 AM" width="819" src="https://user-images.githubusercontent.com/51752063/280452082-3ec90545-9a57-4982-997b-75be8aaf9dab.png">
> for San Diego MSA, the following code returns geoid.
> 
> sd.pop <- get_acs( geography = "tract", variables = "B01003_001", state = state.fips, county = county.fips, geometry = TRUE ) %>% select( GEOID, estimate ) %>% rename( POP=estimate )

Yes, the census.dat does have the tractid. But it does not have GEOID.

**Comment by DrKBoucher** [comment]
> Hi, I'm having the same issue. Census.dat has tractid only while msp.pop1 has GEOID only. So after they merge, my new dataset has 0 rows. Not sure if I did something wrong.

Same exact for me.

**Comment by M-Gamez** [comment]
> > Hi, I'm having the same issue. Census.dat has tractid only while msp.pop1 has GEOID only. So after they merge, my new dataset has 0 rows. Not sure if I did something wrong.
> 
> Having this problem as well

**Comment by Leah-Coffin** [comment]
@AntJam-Howell I have the same issue when I merge the census.dat with the msp.pop I end up with 0 observations of 178 variables.
When I try to run the following: 
# convert sf map object to an sp version
msp.sp <- as_Spatial( msp, cast=TRUE )

I get the error: Error in (function (cond)  : 
  error in evaluating the argument 'x' in selecting a method for function 'addAttrToGeom': subscript out of bounds

From my digging I think it has to do with there be a 0 in the data, but I'm not sure how to omit that; also wondering if it has anything to do with the ocean being a border.

**Comment by AntJam-Howell** [comment]
@Leah-Coffin  Exactly!  If you notice the GEOID and tractid entries look almost identical, excpt for the extra zero placeholder in the front of the GEOID variable.  need to use regex to remove the zero from GEOID so that it will have same format as tractid to make the merge. something like below may work

    # recode the GEOID variable to conform with the census data
    # remove the leading zero
    sd.pop$GEOID<-sub( ".","", sd.pop$GEOID )
